### PR TITLE
begin refactoring to allow passing array of requests

### PR DESCRIPTION
**Pull request by thekemkid**

### CODE REVIEW

**Comment by mcollina** [comment]
I would make the buildRequest a function wrapping the opts in

**Comment by mcollina** [comment]
Sorry about that, I'm on mobile and I pushed the wrong button :/.

Forget about that comment, is perf staying the same?

**Comment by mcollina** [comment]
I would like the builder to have state, meaning it's the builder that maintains the request to be sent.

**Comment by thekemkid** [comment]
Perf is the same :) I'll modify and make this work as appropriate

**Comment by mcollina** [comment]
unit tests? :)

File: lib/httpClient.js [hunk 1]
```diff
@@ -130,24 +121,6 @@ Client.prototype.destroy = function () {
...
-  request.body = newBody || new Buffer(0)
-
-  this._rebuild()
-}
-
-Client.prototype.setRequestHeadersAndBody = function (request, newHeaders, newBody) {
-  request.headers = newHeaders || {}
-  request.body = newBody || new Buffer(0)
-
-  this._rebuild()
```
Comment (mcollina) [inline lib/httpClient.js]: You can call the iterator from these methods.

File: lib/run.js [hunk 2]
```diff
@@ -5,6 +5,18 @@ const URL = require('url')
...
+
+const defaultOptions = {
+  headers: {},
+  body: new Buffer(0),
+  method: 'GET',
+  duration: 10,
+  connections: 10,
+  pipelining: 1,
+  timeout: 10,
+  requests: [{}]
```
Comment (mcollina) [inline lib/run.js]: I don't think that this default is correct, `null` might be better.

File: lib/httpClient.js (hunk 1 above)
Comment (thekemkid) [inline lib/httpClient.js]: I actually started that code, but have a failing test. Will put it in :)

File: lib/run.js (hunk 2 above)
Comment (thekemkid) [inline lib/run.js]: I don't think it is, if the users just pass headers and a body I build it into a single request object in the array, which makes the requests collection usable for a single or multiple requests

**Comment by thekemkid** [comment]
I need unit tests and documentation! I do plan on working on it :D

**Comment by mcollina** [comment]
Great, good work!

**Pull request merged**